### PR TITLE
Add macOS and windows

### PR DESCRIPTION
**Pull request by robertodr**
- [X] Expand OS testing matrix to macOS and Windows. The workflow is run on push *and* on pull requests.
- [x] Hook up to CTest.

### CODE REVIEW

**Comment by robertodr** [comment]
@scanberg I'm reaching the limits of my C :smiley_cat: This happens locally and in the workflow. The headers defining `O_DIRECT` look to be correctly included though :thinking: 

```text
/home/runner/work/mold/mold/src/ext/dcd/fastio.h: In function ‘fio_open’:
/home/runner/work/mold/mold/src/ext/dcd/fastio.h:391:14: error: ‘O_DIRECT’ undeclared (first use in this function); did you mean ‘FIO_DIRECT’?
  391 |     oflag |= O_DIRECT;
      |              ^~~~~~~~
      |              FIO_DIRECT
```

**Comment by scanberg** [comment]
@robertodr I fixed this GNU linux specific issue along alot of other issues. Merge the latest master branch into your fork and see if that resolves all the things!

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -50,6 +43,6 @@ endif()
 if (CMAKE_COMPILER_IS_GNUCC)
 set_property(TARGET mold PROPERTY C_STANDARD 99)
 set_property(TARGET mold PROPERTY C_STANDARD_REQUIRED ON)
```
Comment (robertodr) [inline src/CMakeLists.txt:45]: Should the standard be set to C99 on all platforms? Then on GNU you might enable extensions with `C_EXTENSIONS`?

**Comment by robertodr** [comment]
Lines 215-217 of `md_draw.c` are erroring on macOS. Not sure what to do there.

**Comment by robertodr** [comment]
It seems to work on Linux and Windows, including running the tests with CTest. On macOS, it fails to compile.

**Review by scanberg — APPROVED** [review]
Looks good.
I'll look into the macOS-specific issues, which are due to the static_assert. This could be resolved by bumping the required C version to C11, but I want to see if there is another way of doing it first.

**Comment by scanberg** [comment]
I made some changes to the files (not anything to cmakelists). It compiles locally now on my machine with MSVC, GCC and CLANG. So I hope this resolves the issue with the macos-build failing.

**Comment by robertodr** [comment]
All lanes are green now! :green_apple:

**Pull request merged**